### PR TITLE
feat: add rtc controller

### PR DESCRIPTION
**Pull request by tkhmy**
## PR Type
- New Feature

## Related Links
https://github.com/tier4/autoware_launch/pull/305

## Description
- Handle autoware cooperate commands and status

## TODO
- clean the code, change to class method

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s).
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: autoware_iv_external_api_adaptor/src/rtc_controller.cpp [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/blind_spot/cooperate_status",
+    rclcpp::QoS(1), std::bind(&RTCController::blindSpotCallback, this, _1));
+  crosswalk_sub_ = create_subscription<CooperateStatusArray>(
+    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/crosswalk/cooperate_status",
+    rclcpp::QoS(1), std::bind(&RTCController::crosswalkCallback, this, _1));
+  detection_area_sub_ = create_subscription<CooperateStatusArray>(
+    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/detection_area/cooperate_status",
+    rclcpp::QoS(1), std::bind(&RTCController::detectionAreaCallback, this, _1));
+  intersection_sub_ = create_subscription<CooperateStatusArray>(
+    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/intersect/cooperate_status",
```
Comment (rej55) [inline autoware_iv_external_api_adaptor/src/rtc_controller.cpp]:
```suggestion
    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/intersection/cooperate_status",
```

File: autoware_iv_external_api_adaptor/src/rtc_controller.cpp [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/blind_spot/cooperate_commands",
+    rmw_qos_profile_services_default);
+  cli_set_crosswalk_ = proxy.create_client<CooperateCommands>(
+    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/crosswalk/cooperate_commands",
+    rmw_qos_profile_services_default);
+  cli_set_detection_area_ = proxy.create_client<CooperateCommands>(
+    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/detection_area/cooperate_commands",
+    rmw_qos_profile_services_default);
+  cli_set_intersection_ = proxy.create_client<CooperateCommands>(
+    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/intersect/cooperate_commands",
```
Comment (rej55) [inline autoware_iv_external_api_adaptor/src/rtc_controller.cpp]:
```suggestion
    BEHAVIOR_PLANNING_NAMESPACE + "/behavior_velocity_planner/intersection/cooperate_commands",
```

**Review by rej55 — COMMENTED** [review]
@tkhmy 
Could you check my change suggestion?

**Comment by tkhmy** [comment]
> @tkhmy Could you check my change suggestion?

@rej55 , Thanks! I fixed the naming error

**Review by rej55 — APPROVED** [review]
LGTM

**Pull request merged**